src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -163,4 +171,40 @@ private static void buildThrowable(Throwable t, XContentBuilder builder) throws
             builder.endObject();
         }
     }
-}
\ No newline at end of file
+
+    /*
+     * Builds a simple error response that only will print out the message of the first ElasticsearchException
+     */
+    private static void buildSimpleErrorField(RestChannel channel, Throwable t, XContentBuilder builder) throws IOException {
+        int counter = 0;
+        boolean added = false;
+        Throwable next = t;
+        while (next != null) {
+            if (counter++ > 10) {
+                break;
+            }
+
+            if (t instanceof ElasticsearchException) {
+                builder.field("error", next.getClass().getSimpleName() + "[" + next.getMessage() + "]");
+                added = true;
+                break;
+            }
+            next = next.getCause();
+        }
+
+        if (!added) {
```

Comment:
Perhaps this function would be easier if it were structured like `detailedMessage(t)`? Then this function can just return when it finds an ES exc, and exhausting the loop can have a single return statement at the end.  I think the null case should be extracted out since it is the same whether we do simple or detailed exceptions? The caller locations in convert will look more inline then, like:

```
if (t == null) {
    buidler.field("error", "Unknown");
} else if (channel.emitStackTrace()) {
    builder.field("error", detailedMessage(t));
} else {
    builder.field("error", simpleMessage(t));
}
```